packages/x-data-grid/src/hooks/utils/useGridAriaAttributes.tsx

```diff
@@ -2,30 +2,32 @@ import { gridVisibleColumnDefinitionsSelector } from '../features/columns/gridCo
 import { useGridSelector } from './useGridSelector';
 import { useGridRootProps } from './useGridRootProps';
 import { gridColumnGroupsHeaderMaxDepthSelector } from '../features/columnGrouping/gridColumnGroupsSelector';
-import {
-  gridPinnedRowsCountSelector,
-  gridRowCountSelector,
-} from '../features/rows/gridRowsSelector';
+import { gridPinnedRowsCountSelector } from '../features/rows/gridRowsSelector';
 import { useGridPrivateApiContext } from './useGridPrivateApiContext';
 import { isMultipleRowSelectionEnabled } from '../features/rowSelection/utils';
+import { gridExpandedRowCountSelector } from '../features/filter/gridFilterSelector';
 
 export const useGridAriaAttributes = () => {
   const apiRef = useGridPrivateApiContext();
   const rootProps = useGridRootProps();
   const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
-  const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
+  const accessibleRowCount = useGridSelector(apiRef, gridExpandedRowCountSelector);
   const headerGroupingMaxDepth = useGridSelector(apiRef, gridColumnGroupsHeaderMaxDepthSelector);
   const pinnedRowsCount = useGridSelector(apiRef, gridPinnedRowsCountSelector);
 
   let role = 'grid';
-  if ((rootProps as any).treeData) {
+  if (
+    (rootProps as any).treeData ||
+    (rootProps as any).rowGroupingModel ||
+    (rootProps as any).initialState?.rowGrouping
```

Comment:
Why do we check the initial state? Would it be possible to cause a bug if the initial state has a row grouping entry, but later on the row grouping is removed?

Reply:
Took the first approach, since you will need it in your PR as well

Reply:
At the end, I took another approach
I am using the configuration that has been added to have different row aria attributes for the grid aria as well.
This removes the need to have interfaces and selectors from different grid packages.
Maybe something to consider in the PR you have linked above?